Jenkinsfile

```diff
@@ -1,57 +1,85 @@
 #! groovy
 library 'pipeline-library'
 def nodeVersion = '8.9.0'
-def npmVersion = '5.7.1' // We can change this without any changes to Jenkins. 5.7.1 is minimum to use 'npm ci'
+def npmVersion = '6.8.0' // We can change this without any changes to Jenkins. 5.7.1 is minimum to use 'npm ci'
+
+def packageVersion = ''
+def isPR = env.BRANCH_NAME.startsWith('PR-')
+def runDanger = isPR
+
+def unitTests(titaniumBranch, nodeVersion, npmVersion) {
+	checkout scm
+	nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+		ansiColor('xterm') {
+			timeout(55) {
+				ensureNPM(npmVersion)
+				sh 'npm ci'
+
+				// Install titanium cli
+				ensureNPMPackageVersion('titanium')
+
+				// Install titanium SDK and select it
+				if ('GA'.equals(titaniumBranch)) {
+					sh 'ti sdk install latest -d'
+				} else if ('master'.equals(titaniumBranch)) {
+					sh 'ti sdk install -b master -d'
+				}
+				try {
+					withEnv(["PATH+ALLOY=${pwd()}/bin"]) {
+						sh 'npm test'
+					}
+				} finally {
+					junit 'TEST-*.xml'
+					stash includes: 'TEST-*.xml', name: "test-report-${titaniumBranch}"
+				}
+				
+				fingerprint 'package.json'
+			} // timeout
+		} // ansi
+	} // nodejs
```

Comment:
I think you'd want to run `deleteDir()` at the end here. You have to be careful when you're splitting tasks out in parallel with Jenkins, you really want to try and explicitly clean up whenever you're grabbing a new node and finished with it. Otherwise you may leave behind outdated code/files/generated files.

Reply:
Added this [here](https://github.com/ewanharris/alloy/blob/966becacd88eb2a41985571766dd8ba3cfc3a232/Jenkinsfile#L42)